src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -36,351 +43,670 @@
+				}
+				// lets look it up again to be sure
+				parent = activeJenkins.getItemByFullName(namespace);
+				if (parent instanceof ItemGroup) {
+					return (ItemGroup) parent;
+				}
+			}
+		}
+		return activeJenkins;
+	}
+
+	/**
+	 * Finds the Jenkins job display name for the given {@link BuildConfig}.
+	 *
+	 * @param bc
+	 *            the BuildConfig
+	 * @return the jenkins job display name for the given BuildConfig
+	 */
+	public static String jenkinsJobDisplayName(BuildConfig bc) {
+		String namespace = bc.getMetadata().getNamespace();
+		String name = bc.getMetadata().getName();
+		return jenkinsJobDisplayName(namespace, name);
+	}
+
+	/**
+	 * Creates the Jenkins Job display name for the given buildConfigName
+	 *
+	 * @param namespace
+	 *            the namespace of the build
+	 * @param buildConfigName
+	 *            the name of the {@link BuildConfig} in in the namespace
+	 * @return the jenkins job display name for the given namespace and name
+	 */
+	public static String jenkinsJobDisplayName(String namespace, String buildConfigName) {
+		return namespace + "/" + buildConfigName;
+	}
+
+	/**
+	 * Gets the current namespace running Jenkins inside or returns a reasonable
+	 * default
+	 *
+	 * @param configuredNamespaces
+	 *            the optional configured namespace(s)
+	 * @param client
+	 *            the OpenShift client
+	 * @return the default namespace using either the configuration value, the
+	 *         default namespace on the client or "default"
+	 */
+	public static String[] getNamespaceOrUseDefault(String[] configuredNamespaces, OpenShiftClient client) {
+		String[] namespaces = configuredNamespaces;
+
+		if (namespaces != null) {
+			for (int i = 0; i < namespaces.length; i++) {
+				if (namespaces[i].startsWith("${") && namespaces[i].endsWith("}")) {
+					String envVar = namespaces[i].substring(2, namespaces[i].length() - 1);
+					namespaces[i] = System.getenv(envVar);
+					if (StringUtils.isBlank(namespaces[i])) {
+						logger.warning("No value defined for namespace environment variable `" + envVar + "`");
+					}
+				}
+			}
+		}
+		if (namespaces == null || namespaces.length == 0) {
+			namespaces = new String[] { client.getNamespace() };
+			if (StringUtils.isBlank(namespaces[0])) {
+				namespaces = new String[] { OPENSHIFT_DEFAULT_NAMESPACE };
+			}
+		}
+		return namespaces;
+	}
+
+	/**
+	 * Returns the public URL of the given service
+	 *
+	 * @param openShiftClient
+	 *            the OpenShiftClient to use
+	 * @param defaultProtocolText
+	 *            the protocol text part of a URL such as <code>http://</code>
+	 * @param namespace
+	 *            the Kubernetes namespace
+	 * @param serviceName
+	 *            the service name
+	 * @return the external URL of the service
+	 */
+	public static String getExternalServiceUrl(OpenShiftClient openShiftClient, String defaultProtocolText,
+			String namespace, String serviceName) {
```

Comment:
fyi ... as I mentioned in the proxy url issue ...given that bug my team just got, I am going to add logic which uses the mounted namespace if we are in a pod before using the namesapce from the build

later this quarter we are going to make the jenkins service name configurable.